src/main/java/com/ausregistry/jtoolkit2/se/DomainAdd.java

```diff
@@ -37,5 +37,33 @@ public DomainAdd(String[] nameservers, String[] techContacts,
         super(AddRemType.ADD, nameservers, techContacts, adminContacts,
                 billingContacts, statuses);
     }
+
+    /**
+     * Add associations to a domain.
+     *
+     * @param nameservers A list of host as attribute with the associated domain
+     * To not add any nameservers, this parameter must be null.
+     *
+     * @param techContacts A list of technical contact identifiers to associate
+     * with a domain.  To not add any technical contacts, this parameter must
+     * be null.
+     *
+     * @param adminContacts A list of admin contact identifiers to associate
+     * with a domain.  To not add any admin contacts, this parameter must be
+     * null.
+     *
+     * @param billingContacts A list of billing contact identifiers to
+     * associate with a domain.  To not add any billing contacts, this
+     * parameter must be null.
+     *
+     * @param statuses A list of statuses to add to a domain.  To not add any
+     * statuses, this parameter must be null.
+     */
+    public DomainAdd(Status[] statuses, String[] techContacts,
+        String[] adminContacts, String[] billingContacts, Host[] nameservers) {
```

Comment:
In general it's preferred to keep the parameter list in a similar order in a overloaded methods. Otherwise it will add unnecessary complexity for the user of the toolkit, because they'll have to differentiate the order of parameters based on which version they want to use.

There it'll be better to swap the first and last parameter in the new constructor (see below).
```java
    public DomainAdd(Host[] nameservers, String[] techContacts,
        String[] adminContacts, String[] billingContacts, Status[] statuses) {
```

Reply:
Please refer to my explanation for DomainRem below